zcash_client_sqlite/src/chain.rs

```diff
@@ -58,8 +58,18 @@ where
         ])
         .map_err(to_chain_error)?;
 
+    // Only look for the `from_height` in the scanned blocks if it is set.
+    let mut from_height_found = from_height.is_none();
     while let Some(row) = rows.next().map_err(to_chain_error)? {
         let height = BlockHeight::from_u32(row.get(0).map_err(to_chain_error)?);
+        if !from_height_found {
+            // We will only perform this check on the first row.
+            if from_height != Some(height) {
+                break;
```

Comment:
nonblocking: I slightly prefer to return the error from here directly, and put the `from_height_found = true;` into an `else`.

Reply:
We need to return an error in two places: here, and in the case where there are no rows at all (i.e. `from_height` was set to a height above any block in the database). To avoid a second SQL query (and the transactional logic associated with it), while keeping the error logic simple, I wrote it this way. But with the other proposed change, the inline errors might be straightforward enough that we can duplicate them.

Reply:
Ah, okay, avoiding the duplication is better then.